packages/web3-eth-personal/test/integration/personal.test.ts

```diff
@@ -0,0 +1,125 @@
+/*
+This file is part of web3.js.
+
+web3.js is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+web3.js is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
+*/
+import { isHexStrict } from 'web3-validator';
+import { toChecksumAddress } from 'web3-utils';
+import { EthPersonal } from '../../src/index';
+import { importedAccount } from '../config/personal.test.config';
+import {
+	getSystemTestBackend,
+	getSystemTestAccounts,
+	getSystemTestProvider,
+	itIf,
+} from '../fixtures/system_test_utils';
+
+describe('personal integration tests', () => {
+	let ethPersonal: EthPersonal;
+	let accounts: string[];
+	beforeAll(() => {
+		const clientUrl = getSystemTestProvider();
+		ethPersonal = new EthPersonal(clientUrl);
+	});
+	beforeEach(async () => {
+		accounts = await getSystemTestAccounts();
+	});
+	it('new account', async () => {
+		const newAccount = await ethPersonal.newAccount('!@superpassword');
+		expect(isHexStrict(newAccount)).toBe(true);
+	});
+
+	itIf(getSystemTestBackend() === 'geth')('ecRecover', async () => {
+		// ganache does not support ecRecover
+		const signature = await ethPersonal.sign('0x2313', accounts[0], '');
+		const publicKey = await ethPersonal.ecRecover('0x2313', signature); // ecRecover is returning all lowercase
+		// eslint-disable-next-line jest/no-standalone-expect
+		expect(toChecksumAddress(publicKey)).toBe(toChecksumAddress(accounts[0]));
+	});
+
+	it('lock account', async () => {
+		const lockAccount = await ethPersonal.lockAccount(accounts[1]);
+		expect(lockAccount).toBe(true);
+	});
```

Comment:
Should we not test if the account is actually locked for it? What if the implementation of `lockAccount` function simply says `return true`, this test will be a false positive in that case.

Reply:
From what I understand, there is no direct command to see if an account is locked/unlocked other than sending a transaction and seeing if errors, which is something i can definitely do.

Reply:
Yes that's called the [side effect](https://en.wikipedia.org/wiki/Side_effect_(computer_science)). 

> an observable effect besides returning a value (the primary effect) to the invoker of the operation

If there is only way to test the completion of operation through it's side effects then we should test it as part of the integration/system tests.